### PR TITLE
Part3 alter

### PR DESCRIPTION
**Pull request by omerporze**

### CODE REVIEW

File: part3/L4-normal.test.ts [hunk 1]
```diff
@@ -94,7 +94,8 @@ describe('L4 Normal Eval', () => {
                             (lambda (x) (f ((nf f (- n 1)) x)))))))
                 ((nf (lambda (x) (* x x)) 2) 3))`), evalNormalProgram)).to.deep.equal(makeOk(81));
 
-        // Accidental capture of the z variable if no renaming
+        // Accidental capture of the z variable if no renaming - TODO: this means renaming is a must, so should be taken from other branch?
```
Comment (IdanTo1) [inline part3/L4-normal.test.ts:97]: TODO?

File: part3/L4-normal.ts [hunk 2]
```diff
@@ -4,18 +4,88 @@ import { Sexp } from "s-expression";
...
-import { isTrueValue } from "./L4-eval";
+         isPrimOp, isProcExp, isStrExp, isVarRef, VarDecl} from "./L4-ast";
+import { applyEnv, makeEmptyEnv, Env , makeExtEnv, makeRecEnv} from './L4-env-normal';
 import { applyPrimitive } from "./evalPrimitive";
 import { isClosure, makeClosure, Value } from "./L4-value";
 import { first, rest, isEmpty } from '../shared/list';
 import { Result, makeOk, makeFailure, bind, mapResult } from "../shared/result";
 import { parse as p } from "../shared/parser";
 
+export const normalEval = (exp: CExp, env: Env): Result<Value> =>
```
Comment (IdanTo1) [inline part3/L4-normal.ts:15]: align everything to same line, as they do in their examples

File: part3/L4-value.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@
 // Value type definition for L4
 
 import { isPrimOp, CExp, PrimOp, VarDecl } from './L4-ast';
-import { Env } from './L4-env';
+import { Env } from './L4-env-normal';
```
Comment (IdanTo1) [inline part3/L4-value.ts:5]: it is their change?

File: part3/L4-normal.test.ts (hunk 1 above)
Comment (omerporze) [reply part3/L4-normal.test.ts:97]: ohh forgot it there

File: part3/L4-normal.ts (hunk 2 above)
Comment (omerporze) [reply part3/L4-normal.ts:15]: done

File: part3/L4-value.ts (hunk 3 above)
Comment (omerporze) [reply part3/L4-value.ts:5]: mine, we need to use new env

**Pull request merged**